wolfcrypt/test/test.c

```diff
@@ -1980,6 +1980,15 @@ WOLFSSL_TEST_SUBROUTINE int md2_test(void)
         if (XMEMCMP(hash, test_md2[i].output, MD2_DIGEST_SIZE) != 0)
             return -1500 - i;
     }
+    
+    for (i = 0; i < times; ++i){
+        wc_Md2Hash((byte*)test_md2[i].input, (word32)test_md2[i].inLen, hash);
+        
+        if (XMEMCMP(hash, test_md2[i].output, MD2_DIGEST_SIZE) != 0){
+            return -1500 - i;
```

Comment:
"-1500 - i" is used in the above test block.  We should start this one at a later offset (larger negative than -1500, so in the case this fails, we will be able to tell it failed in this for() instead of the above one.

Reply:
Let's either move the (ret != 0) check above your XMEMCMP, or put them in the same if with &&.